deploifai/api/__init__.py

```diff
@@ -410,3 +409,213 @@ def create_project(self, project_name: str, cloud_profile: CloudProfile, fragmen
+            "usesGpu": uses_gpu,
+            "whereProvider": cloud_provider
+        }
+
+        try:
+            r = requests.post(
+                self.uri,
+                json={"query": query, "variables": variables},
+                headers=self.headers,
+            )
+            api_data = r.json()
+
+            return api_data["data"]["trainingInfrastructurePlans"]
+        except TypeError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+        except KeyError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+
+    def falcon_ml_config(self, variable):
+        query = (
+            """    
+            query ($where: FalconMLConfigWhereInput)
+                {
+                    falconMLConfigs(where: $where){
+                        id
+                        pythonVersion
+                        frameworkVersion
+                        cudaVersion
+                        cudnnVersion
+                    }
+                }
+            """
+        )
+        variables = {
+            "where": variable
+        }
+
+        try:
+            r = requests.post(
+                self.uri,
+                json={"query": query, "variables": variables},
+                headers=self.headers,
+            )
+            api_data = r.json()
+
+            return api_data["data"]["falconMLConfigs"]
+        except TypeError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+        except KeyError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+
+    def falcon_ml_config_distinct(self, variable, distinct_option: str):
+        query = (
+            """    
+            query ($where: FalconMLConfigWhereInput $distinct: [FalconMLConfigScalarFieldEnum!])
+                {
+                    falconMLConfigs(where: $where distinct: $distinct){
+                        id
+                        pythonVersion
+                        frameworkVersion
+                        cudaVersion
+                        cudnnVersion
+                    }
+                }
+            """
+        )
+        variables = {
+            "where": variable,
+            "distinct": [distinct_option],
+        }
+
+        try:
+            r = requests.post(
+                self.uri,
+                json={"query": query, "variables": variables},
+                headers=self.headers,
+            )
+            api_data = r.json()
+
+            return api_data["data"]["falconMLConfigs"]
+        except TypeError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+        except KeyError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+
+    def create_training_server(self, name: str, project_id: str, cloud_profile_id: str,
+                               falcon_plan: str, falcon_gpu: str,
```

Comment:
`falcon_gpu` argument should have been a `bool` type instead of `str`. This still works during runtime because python doesn't really check static types. But its important to be clear for internal documentation. 

And also should name the variable as `uses_gpu` to be consistent. It's better to have longer variable names, even though they can be a bit annoying at first, but self-explanatory variable naming is super beneficial for the long term. Could make some modifications to `falcon_id` into `falcon_ml_config_id` instead.

Reply:
refactored